Formula/a/apache-arrow-glib.rb

```diff
@@ -31,6 +32,15 @@ class ApacheArrowGlib < Formula
   fails_with gcc: "5"
 
   def install
+    # Work around an Xcode 15 linker issue which causes linkage against LLVM's
+    # libunwind due to it being present in a library search path.
+    if DevelopmentTools.clang_build_version >= 1500
+      recursive_dependencies
+        .select { |d| d.name.match?(/^llvm(@\d+)?$/) }
+        .map { |llvm_dep| llvm_dep.to_formula.opt_lib }
+        .each { |llvm_lib| ENV.remove "HOMEBREW_LIBRARY_PATHS", llvm_lib }
+    end
```

Comment:
Should probably just handle this in `brew` at this point.

Though, dare I ask -- why don't we want linkage with our `libunwind`? As long as they're all using the same one it should be ok... (Though admittedly guaranteeing that may be a tall order)

Reply:
> Should probably just handle this in `brew` at this point.

Yea, I agree. Was looking into this too.

> Though, dare I ask -- why don't we want linkage with our `libunwind`? As long as they're all using the same one it should be ok... (Though admittedly guaranteeing that may be a tall order)

Because we cannot be sure the LLVM used to build the formula (or simply present at build time) will continue to stay around until the next rebuild/update -- unless it's a direct runtime dependency.

Some formulae have a build-only dependency on `llvm`. The linkage test would have failed should there be libunwind linkage. This was the case when this workaround was added to `apache-arrow` in #146072 though LLVM became a runtime dependency later.

Some others, like the dependents here, depend on `llvm` indirectly via another runtime dependency (`apache-arrow` in this case). This makes it unnecessarily challenging for us to switch the dependent formula to a different `llvm`. Even a "hello world" can link to libunwind unexpectedly.

```console
$ cat test.cc
#include <iostream>

int main() {
  std::cout << "hello, world\n";
}
$ LIBRARY_PATH="$(brew --prefix llvm)/lib" "$(brew --prefix llvm)/bin/clang++" test.cc
ld: warning: reexported library with install name '/usr/local/opt/llvm/lib/libunwind.1.dylib' found at '/usr/local/Cellar/llvm/17.0.6_1/lib/libunwind.1.0.dylib' couldn't be matched with any parent library and will be linked directly
$ otool -L ./a.out
./a.out:
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1700.255.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1345.100.2)
	/usr/local/opt/llvm/lib/libunwind.1.dylib (compatibility version 1.0.0, current version 1.0.0)
```

Reply:
Homebrew/brew#17104

Reply:
> Even a "hello world" can link to libunwind unexpectedly.
> 
> ```
> $ cat test.cc
> #include <iostream>
> 
> int main() {
>   std::cout << "hello, world\n";
> }
> $ LIBRARY_PATH="$(brew --prefix llvm)/lib" "$(brew --prefix llvm)/bin/clang++" test.cc
> ld: warning: reexported library with install name '/usr/local/opt/llvm/lib/libunwind.1.dylib' found at '/usr/local/Cellar/llvm/17.0.6_1/lib/libunwind.1.0.dylib' couldn't be matched with any parent library and will be linked directly
> $ otool -L ./a.out
> ./a.out:
> 	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1700.255.0)
> 	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1345.100.2)
> 	/usr/local/opt/llvm/lib/libunwind.1.dylib (compatibility version 1.0.0, current version 1.0.0)
> ```

This is really weird. Smells like a bug in `ld` to me. Thoughts, @Bo98?